src/webgl/shaders/lighting.glsl

```diff
@@ -67,6 +79,96 @@ LightResult _light(vec3 viewDirection, vec3 normal, vec3 lightVector) {
   return lr;
 }
 
+// converts the range of "value" from [min1 to max1] to [min2 to max2]
+float map(float value, float min1, float max1, float min2, float max2) {
+  return min2 + (value - min1) * (max2 - min2) / (max1 - min1);
+}
+
+/*
+vec2 mapTextureToNormal( vec3 normal ){
+  // x = r sin(phi) cos(theta)
+  // y = r cos(phi)
+  // z = r sin(phi) sin(theta)
+  float phi = acos( normal.y );
+  // if phi is 0, then there are no x, z components
+  float theta = 0.0;
+  // else 
+  if(  abs(sin(phi)) > 0.0 ){
+    theta = acos( normal.x / sin(phi) );
+  }   
+
+  vec2 newTexCoor = vec2(
+    map(theta, 0., PI, 0., 1.),
+    map(phi, 0., PI, 1., 0.)
+  );
+
+  return newTexCoor;
+}
+*/
+
+vec2 nToE( vec3 v ){
+  // x = r sin(phi) cos(theta)   
+  // y = r cos(phi)  
+  // z = r sin(phi) sin(theta)
+  float phi = acos( v.y );
+  // if phi is 0, then there are no x, z components
+  float theta = 0.0;
+  // else 
+  theta = acos(v.x / sin(phi));
+  float sinTheta = v.z / sin(phi);
+  if (sinTheta < 0.0) {
+    // Turn it into -theta, but in the 0-2PI range
+    theta = 2.0 * PI - theta;
+  }
+  theta = theta / (2.0 * 3.14159);
+  phi = phi / 3.14159 ;
+  
+  vec2 angles = vec2( phi, theta );
```

Comment:
Looks like something might be up still:
![image](https://github.com/processing/p5.js/assets/5315059/0f6ca301-86f7-4a63-bad9-bb0028c07eab)

I noticed that we had a note for diffused light to use the word normal instead of `R`. When I use that, it looks like this:
![image](https://github.com/processing/p5.js/assets/5315059/68955a40-ce1c-4b4e-b6d1-bce474eb9dbf)

I tried using a two-toned background to check against and it seems like it's kind of flipping it:
![image](https://github.com/processing/p5.js/assets/5315059/8862c67d-4b04-4a2d-91b2-ae43fb55e5c5)


That looks a bit more correct, but it looks like the green is on the side? I think that might be because in other files, we used theta and phi differently, but here, we want to return `vec2(theta, phi)` instead of the other way around (I think that makes sense that the x axis is theta here since that's the one whose range is 2*PI. That's different than in the other shaders, which is a little odd, but at least we tested those independently and they seem fine.) That looks like this:

![image](https://github.com/processing/p5.js/assets/5315059/74d75ff8-65c4-4b35-b6ac-f8ed9b6e080a)

so that still seems 90 degrees off, but in a different axis this time. I tried cycling -90 degrees in the theta axis. This seems to look right, and that's using `vec2 angles = vec2( fract(theta + 0.75), 1.0 - phi );`
![image](https://github.com/processing/p5.js/assets/5315059/6e1c9d0a-b2b0-488c-94e5-610a26d1ade3)

With just diffuse: 
![image](https://github.com/processing/p5.js/assets/5315059/7a714e26-5e04-4273-bd41-752bce061d47)

With specular:
![image](https://github.com/processing/p5.js/assets/5315059/62841b14-96a0-4c94-85cf-5b3ed1578a38)

```suggestion
  vec2 angles = vec2( fract(theta + 0.75), 1.0 - phi );
```

Reply:
Changed this line, but do i have to change the line 
```
  // use worldNormal instead of R
  vec2 newTexCoor = mapTextureToNormal( R );
```
Did you use worldNormal or R in this function?

Reply:
> Did you use worldNormal or R in this function?

I think that one should be `worldNormal`.

I double checked the suggested code and I think maybe I was testing without that by accident though -- I think when using the world normal, to get the example with the stripes to work, it would be this:

```glsl
vec2 angles = vec2( fract(theta + 0.25), 1.0 - phi );
```

(Basically, 0.25 instead of 0.75, which is like rotating by pi.)